core/modules/replica/WorkerServerConnection.cc

```diff
@@ -0,0 +1,543 @@
+                          _bufferPtr->parseLength(),
+                          hdr)) return;
+
+    // Analyse the header of the request. Note that the header message categorizes
+    // requests in two layers:
+    // - first goes the class of requests as defined by member 'type'
+    // - then  goes a choice of a specific request within its class. Those specific
+    //   request codes are obtained from the corresponding members
+
+    switch (hdr.type()) {
+
+        case proto::ReplicationRequestHeader::REPLICA: processReplicaRequest(   hdr); break;
+        case proto::ReplicationRequestHeader::REQUEST: processManagementRequest(hdr); break;
+        case proto::ReplicationRequestHeader::SERVICE: processServiceRequest(   hdr); break;
+
+        default:
+            throw std::logic_error(
+                  "WorkerServerConnection::received() unhandled request class: '" +
+                  proto::ReplicationRequestHeader::RequestType_Name(hdr.type()));
+    }
+}
+
+void WorkerServerConnection::processReplicaRequest(proto::ReplicationRequestHeader& hdr) {
+
+    // Read the request length
+    uint32_t bytes;
+    if (not ::readLength (_socket,
+                          _bufferPtr,
+                          bytes)) return;
+
+    switch (hdr.replica_type()) {
+
+        case proto::ReplicationReplicaRequestType::REPLICA_CREATE: {
+
+            // Read the request body
+            proto::ReplicationRequestReplicate request;
+            if (not ::readMessage(_socket,
+                                  _bufferPtr,
+                                  bytes,
+                                  request)) return;
+
+            proto::ReplicationResponseReplicate response;
+            _processor.enqueueForReplication(hdr.id(),
+                                             request,
+                                             response);
+            reply(hdr.id(),
+                  response);
+
+            break;
+        }
+        case proto::ReplicationReplicaRequestType::REPLICA_DELETE: {
+
+            // Read the request body
+            proto::ReplicationRequestDelete request;
+            if (not ::readMessage(_socket,
+                                  _bufferPtr,
+                                  bytes,
+                                  request)) return;
+
+            proto::ReplicationResponseDelete response;
+            _processor.enqueueForDeletion(hdr.id(),
+                                          request,
+                                          response);
+            reply(hdr.id(),
+                  response);
+
+            break;
+        }
+        case proto::ReplicationReplicaRequestType::REPLICA_FIND: {
+
+            // Read the request body
+            proto::ReplicationRequestFind request;
+            if (not ::readMessage(_socket,
+                                  _bufferPtr,
+                                  bytes,
+                                  request)) return;
```

Comment:
I'll stop flagging these now, I assume you know my position 😄

Reply:
expanded